TAX-2153 [new]: Tax Provider API, add performance guidance (#404)
bigcommerce/docs: docs/integrations/tax.mdx

  You may request a lower timeout threshold to provide a better service to merchants.
</Callout>

#### Handling additional inputs

The Tax Provider API's request specification has been designed to include all necessary inputs into a standard tax calculation. If your solution requires additional inputs beyond what the request model offers, then this additional data should be accessible on your system in a performant manner.

In particular this means avoiding synchronous calls to external systems while preparing your tax estimate response. A recommended performant implementation strategy is to maintain a projection of external data on your system, performing updates asynchronously through background jobs or queues as required (e.g. [subscribing to BigCommerce webhooks](https://developer.bigcommerce.com/docs/integrations/webhooks)).

> Leaning on capabilities of [Tax Properties](/docs/store-operations/tax/tax-properties) to enhance the initial Tax Provider API request is another available approach for avoiding the need to look up additional tax estimate inputs on external systems.

Reviewer: `Leaning on the capabilities` reads better to me

> Leaning on capabilities of [Tax Properties](/docs/store-operations/tax/tax-properties) to enhance the initial Tax Provider API request is another available approach for avoiding the need to look up additional tax estimate inputs on external systems.

Author: Thanks @benpratt77, I like it

  You may request a lower timeout threshold to provide a better service to merchants.
</Callout>

#### Handling additional inputs

> The Tax Provider API's request specification has been designed to include all necessary inputs into a standard tax calculation. If your solution requires additional inputs beyond what the request model offers, then this additional data should be accessible on your system in a performant manner.

Reviewer: What would you think of `to include all necessary inputs required to perform a standard tax calculation`

> The Tax Provider API's request specification has been designed to include all necessary inputs into a standard tax calculation. If your solution requires additional inputs beyond what the request model offers, then this additional data should be accessible on your system in a performant manner.

Author: Hmm `necessary` and `required` are a bit duplicative, but you might be onto something with `to perform a`.

> The Tax Provider API's request specification has been designed to include all necessary inputs into a standard tax calculation. If your solution requires additional inputs beyond what the request model offers, then this additional data should be accessible on your system in a performant manner.

Author: I've also switched to `tax estimation` rather than `tax calculation`, better mirroring existing language int his area.

> The Tax Provider API's request specification has been designed to include all necessary inputs required to perform a standard tax estimation. If your solution requires additional inputs beyond what the request model offers, then this additional data should be accessible on your system in a performant manner.

Reviewer 2: ```suggestion
The Tax Provider API's request specification includes all necessary inputs for standard tax estimation. If your solution needs extra inputs beyond the request model, ensure this data is readily accessible on your system.
```

> In particular this means avoiding synchronous calls to external systems while preparing your tax estimate response. A recommended performant implementation strategy is to maintain a projection of external data on your system, performing updates asynchronously through background jobs or queues as required (e.g. [subscribing to BigCommerce webhooks](https://developer.bigcommerce.com/docs/integrations/webhooks)).

Reviewer 2: ```suggestion
Avoid synchronous calls to external systems when preparing your tax estimate response. We recommend you keep a copy of external data on your system and update it asynchronously through background jobs or queues (e.g., [subscribing to BigCommerce webhooks](https://developer.bigcommerce.com/docs/integrations/webhooks)).
```

Author: Fantastic, as always @bc-andreadao.

Only interesting thing here is the `e.g.,` update - on purpose? I'll update the others in this document if so - but keen to understand that update a little more 🙏

Reviewer 2: Thanks @theromulans! Yes, there should be a comma after e.g. 
![Uploading comma.png…]()

Author: TIL, thank you @bc-andreadao!

> Leaning on the capabilities of [Tax Properties](/docs/store-operations/tax/tax-properties) to enhance the initial Tax Provider API request is another available approach for avoiding the need to look up additional tax estimate inputs on external systems.

Reviewer 2: ```suggestion
Using [Tax Properties](/docs/store-operations/tax/tax-properties) to enhance the initial Tax Provider API request can help you avoid looking up additional tax estimate inputs on external systems.
```
